metagraph-plugin/metagraph_katana/tests/test_algorithms.py

```diff
@@ -0,0 +1,247 @@
+def test_pagerank(nx_weighted_directed_8_12, kg_from_nx_di_8_12):
+    pr_nx = mg.algos.centrality.pagerank(nx_weighted_directed_8_12)
+    pr_kg = mg.algos.centrality.pagerank(kg_from_nx_di_8_12)
+    assert isinstance(pr_nx, dict)
+    assert isinstance(pr_kg, dict)
+    assert pr_nx == pr_kg
+    
+
+# betweenness centrality
+# mg.algos.centrality.betweenness(graph: Graph(edge_type=’map’, edge_dtype={‘int’, ‘float’}), nodes: Optional[NodeSet] = None, normalize: bool = False) → NodeMap
+# the results of these 2 approaches are compared:
+# (1) katanagraph->nxgraph + nx's betweenness centrality (2) nxgraph + nx's betweenness centrality
+def test_betweenness_centrality(nx_weighted_directed_8_12, kg_from_nx_di_8_12):
+    bc_nx = mg.algos.centrality.betweenness(nx_weighted_directed_8_12)
+    bc_kg = mg.algos.centrality.betweenness(kg_from_nx_di_8_12)
+    assert isinstance(bc_nx, dict)
+    assert isinstance(bc_kg, dict)
+    assert bc_nx == bc_kg
+
+# triangle counting
+# mg.algos.clustering.triangle_count(graph: Graph(is_directed=False)) → int
+# the results of these 2 approaches are compared:
+# (1) katanagraph->nxgraph + nx's triangle counting (2) nxgraph + nx's triangle counting
+def test_triangle_counting(nx_weighted_undirected_8_12, kg_from_nx_ud_8_12):
+    tc_nx = mg.algos.clustering.triangle_count(nx_weighted_undirected_8_12)
+    tc_kg = mg.algos.clustering.triangle_count(kg_from_nx_ud_8_12)
+    assert isinstance(tc_nx, int)
+    assert isinstance(tc_kg, int)
+    assert tc_nx == tc_kg
+
+
+# Louvain community detection
+# mg.algos.clustering.louvain_community(graph: Graph(is_directed=False, edge_type=’map’, edge_dtype={‘int’, ‘float’})) → Tuple[NodeMap, float]
+# the results of these 2 approaches are compared:
+# (1) katanagraph->nxgraph + nx's Louvain community detection (2) nxgraph + nx's Louvain community detection
+def test_louvain_community_detection(nx_weighted_undirected_8_12, kg_from_nx_ud_8_12):
+    lc_nx = mg.algos.clustering.louvain_community(nx_weighted_undirected_8_12)
+    lc_kg = mg.algos.clustering.louvain_community(kg_from_nx_ud_8_12)
+    assert isinstance(lc_nx[0], dict)
+    assert isinstance(lc_kg[0], dict)
+    assert isinstance(lc_nx[1], float)
+    assert isinstance(lc_kg[1], float)
+    assert lc_nx[0] == lc_kg[0]
+    assert lc_nx[1] == lc_kg[1]
+
+
+# subgraph extraction
+# mg.algos.subgraph.extract_subgraph(graph: Graph, nodes: NodeSet) → Graph
+# the results of these 2 approaches are compared:
+# (1) katanagraph->nxgraph + nx's subgraph extraction (2) nxgraph + nx's subgraph extraction
+def test_subgraph_extraction(nx_weighted_directed_8_12, kg_from_nx_di_8_12):
+    se_nx = mg.algos.subgraph.extract_subgraph(nx_weighted_directed_8_12, {0,2,3})
+    se_kg = mg.algos.subgraph.extract_subgraph(kg_from_nx_di_8_12, {0,2,3})
+    assert isinstance(se_nx, mg.wrappers.Graph.NetworkXGraph)
+    assert isinstance(se_kg, mg.wrappers.Graph.NetworkXGraph)
```

Comment:
Why is the extracted subgraph a NetworkXGraph instead of a new Katana Graph?

Reply:
for algorithm calling network graph's algorithm, it will return the networkx type. If we bound add the plugin katana algorithm for that, it will return a Katana Graph.

Reply:
OK. So this isn't a test of subgraph_extraction. This is a test of graph conversion before using a non-katana algorithm. That's fine. But you should rename this test.

Reply:
They are using the same algorithm interface, and that's why I put them in test_algorithm.py. The test interface does not change if we (in the future add the katana plugin for that algorithm). How about I only change the name of the test function, add a mark of "_translation" or something?

After we add the plugin algorithm, I can change the name back.

Reply:
I would rather have one test of transitioning between graph types automatically that will always test that, even as we add new algorithms. Then we can just remove these tests entirely and reintroduce them bit by bit as we wrap more algorithms. I'm not a fan of having code here that is in some sense not used.